banno-enrollment-po/BANNO.ENROLLMENT.V1.POW

```diff
@@ -0,0 +1,772 @@
+                           PREFERENCE:HBMODE=STDHOMEBANKING)
+  DO
+   PREFCOUNT=PREFCOUNT+1
+   PREFLOC(PREFCOUNT)=PREFERENCE:LOCATOR
+   PREFUSER(PREFCOUNT)=PREFERENCE:HBUSERNAME
+   PREFEMAILADDR(PREFCOUNT)=PREFERENCE:EMAILADDRESS
+   PREFPASSPROB(PREFCOUNT)=PREFERENCE:PWDRESETPROBATION
+   PREFXFERANYLOC(PREFCOUNT)=0
+   FOR EACH PREFERENCE ACCESS WITH (PREFERENCE ACCESS:ACCESSTYPE=3)
+    DO
+     PREFXFERANYLOC(PREFCOUNT)=PREFERENCE ACCESS:LOCATOR
+    END
+   UNTIL PREFXFERANYLOC(PREFCOUNT)<>0
+  END
+ IF PREFCOUNT>PREFMAX THEN
+  DO
+   POPUPMESSAGE(2,"Too many users, Update Manually")
+   TERMINATE
+  END
+END [PROCEDURE]
+
+PROCEDURE CHOOSEACTION
+[* Dialog that prompts user for desired action.
+*]
+ DIALOGSTART("Banno Enrollment",1500%,0)
+ DIALOGPROMPTLISTSTART("Choose Action:",ACTIONCHGPASS)
+ IF PREFCOUNT>=1 THEN
+  DIALOGPROMPTLISTOPTION(ACTIONCHGPASS,"Change Password")
+ IF PREFCOUNT=0 OR ALLOWMULT=TRUE THEN
+  DIALOGPROMPTLISTOPTION(ACTIONCREATEUSR,"Create User")
+ IF PREFCOUNT>=1 THEN
+  DIALOGPROMPTLISTOPTION(ACTIONEDITUSER,"Edit User")
+ IF ALLOWDELETE=TRUE AND PREFCOUNT>=1 THEN
+  DIALOGPROMPTLISTOPTION(ACTIONDELUSER,"Delete User")
+ IF ALLOWXACCT=TRUE AND PREFCOUNT>=1 THEN
+  DO
+   DIALOGPROMPTLISTOPTION(ACTIONADDXACCT,"Add Cross Account")
+   DIALOGPROMPTLISTOPTION(ACTIONEDITXACCT,"Edit Cross Account")
+   DIALOGPROMPTLISTOPTION(ACTIONDELXACCT,"Delete Cross Account")
+  END
+ DIALOGPROMPTLISTOPTION(ACTIONEXITPGM,"Exit Banno Enrollment")
+ DIALOGPROMPTLISTEND
+ DIALOGDISPLAY
+ ACTIONREQUESTED=ENTERNUMBER("Choose Action:",ACTIONCHGPASS)
+ DIALOGCLOSE
+END [PROCEDURE]
+
+PROCEDURE CHGPASSDIALOG
+[* Dialog that prompts user for new HB password and password
+** probation preference.
+*]
+ DIALOGSTART("Banno - New Password",200%,0)
+ DIALOGINTROTEXT("Changing Password for User "+PREFUSER(SELECTEDPREF))
+ DIALOGPROMPTCHAR("Enter New Password:",GETDATANUMBER(GETPARAMHBPASSWORDMAXLENGTH),"")
+ DIALOGPROMPTYESNO("Change Password at next login?",FALSE)
+ DIALOGDISPLAY
+ NEWPASSWORD=ENTERCHARACTER("Enter New Password:",GETDATANUMBER(GETPARAMHBPASSWORDMAXLENGTH),"")
+ NEWPASSPROB=ENTERYESNO("Change Password at next login?","0")
+ DIALOGCLOSE
+END [PROCEDURE]
+
+PROCEDURE CHGPASSFM
+[* File maintenance and error handling for password change.
+*]
+ IF NEWPASSPROB="Y" THEN
+  NEWPASSPROBNUM=1
+ ELSE
+  NEWPASSPROBNUM=0
+
+ FMPERFORM REVISE PREFERENCE LOC PREFLOC(SELECTEDPREF)(CHECKPRIVS,0,FMERROR)
+  DO
+   SET HBPASSWORD TO NEWPASSWORD
+   SET LASTHBPWCHANGEDATE TO SYSTEMDATE
+   SET PWDRESETPROBATION TO NEWPASSPROBNUM
+  END
+ IF FMERROR<>"" THEN
+  CALL ERRORHANDLER
+ ELSE IF FMERROR="" AND NEWPASSWORD<>"" THEN
+  POPUPMESSAGE(0,"Password Successfully Updated")
+ ELSE IF FMERROR="" AND NEWPASSWORD="" THEN
+  POPUPMESSAGE(0,"Password Successfully Blanked Out")
+END [PROCEDURE]
+
+PROCEDURE CREATEEDITDIALOG
+[* Dialog that prompts user for information needed when creating
+** or editing a user.
+*]
```

Comment:
Consider having title reflect what user is doing… ‘add’ or ‘edit’ but not both.

Reply:
Done. Cleaned up all dialog titles